### PR TITLE
replies, forwards, reactions 

### PR DESCRIPTION
**Pull request by sdfsdhgjkbmnmxc**

### CODE REVIEW

File: messages.proto [hunk 1]
```diff
@@ -30,32 +30,64 @@ enum MessageRight {
   CAN_DELETE_MESSAGE = 2;
 }
 
+// Full message with all fields
 message Message {
   ChatPreview chat = 8;
   string id = 1;
   uint64 created_at = 3;
-  uint64 updated_at = 9;
+  uint64 changed_at = 9; // when equal to created_at, message was not edited
```
Comment (Maksimka101) [inline messages.proto]: Why did you decided to rename this field?

File: messages.proto [hunk 2]
```diff
@@ -30,32 +30,64 @@ enum MessageRight {
...
   string text = 4;
   User sender = 5;
   string prev_message_id = 6;
   uint32 viewed_counter = 7; // how many recipients viewed this message
   repeated MessageRight rights = 10;
+
+  // attachments
+  MessageReference reply = 12;
```
Comment (Maksimka101) [inline messages.proto]: What's the difference between forward a single message from the same chat and reply?

Comment (Maksimka101) [inline messages.proto:46]: What's the difference between forward a single message from the same chat and reply?

Comment (Maksimka101) [inline messages.proto:46]: What's the difference between forward a single message from the same chat and reply?

File: messages.proto [hunk 3]
```diff
@@ -30,32 +30,64 @@ enum MessageRight {
...
+  repeated Reaction reactions = 14;
+}
+
+message Reaction {
+  string emoji = 1;
+  repeated User users = 2;
+}
+
+// Short message representation for replies and forwards
+message MessageReference {
```
Comment (Maksimka101) [inline messages.proto]: Maybe CompactMessage?

File: messages.proto (hunk 2 above)
Comment (sdfsdhgjkbmnmxc) [reply messages.proto]: They looks different in UI:

1) reply looks like:

```
| replied message ....(long text truncated)...
message.text
```

2) forwards looks like:
```
message.text
| forwarded message 
| forwarded message
| forwarded message
```

File: messages.proto (hunk 1 above)
Comment (sdfsdhgjkbmnmxc) [reply messages.proto]: * `updated_at` reserved for any update (e.g. reactions count)
* `changed_at` means that message edited by author, it happens only on text / uploads / reply / forwards change

File: messages.proto (hunk 3 above)
Comment (sdfsdhgjkbmnmxc) [reply messages.proto]: ...or `ShortMessage`...

**Pull request merged**